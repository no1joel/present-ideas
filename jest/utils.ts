import {
  render as testingLibraryRender,
  RenderOptions,
} from "@testing-library/vue";
import linkify from "vue-linkify";
import { createMemoryHistory, createRouter } from "vue-router";
import routes from "../present_ideas/present_ideas/static/router/routes";
import getStore from "../present_ideas/present_ideas/static/store";

const mockRouter = createRouter({ routes, history: createMemoryHistory() });

export const render = async (component, options = {}) => {
  const store = getStore();
  const defaultRenderOptions: RenderOptions = {
    global: {
      plugins: [store, mockRouter],
      directives: { linkified: linkify },
    },
  };
  const renderOptionsWithDefaults = { ...defaultRenderOptions, ...options };

  const result = testingLibraryRender(component, renderOptionsWithDefaults);

  return result;
};
