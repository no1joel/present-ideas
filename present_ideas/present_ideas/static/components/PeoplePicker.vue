<template>
  <select>
    <option disabled>Select a person</option>
    <option v-if="loading" disabled>Loading...</option>
    <option v-for="person in people" v-bind:key="person.name">
      {{ person[0].toUpperCase() + person.slice(1) }}
    </option>
  </select>
</template>

<script>
export default {
  data() {
    return {
      people: [],
      loading: true,
    };
  },
  mounted() {
    this.loadPeople();
  },
  methods: {
    async loadPeople() {
      this.loading = true;
      const response = await fetch("/api/person");
      const data = await response.json();
      const names = data.names;
      this.people = names.sort();
      this.loading = false;
    },
  },
};
</script>