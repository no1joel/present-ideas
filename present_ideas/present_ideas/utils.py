from typing import Iterable, List

from sheets.spreadsheet import (
    add_row,
    delete_row,
    get_rows,
    get_worksheets,
    set_cell_value,
)


def check_name(name: str) -> bool:
    """Check if the provided name is a sheet."""

    worksheet_names = get_all_names()
    return name.lower() in worksheet_names


def get_all_names() -> Iterable[str]:
    """Return people's names."""

    return (
        worksheet.title.lower()
        for worksheet in get_worksheets()
        if "template" not in worksheet.title.lower()
    )


def get_present_ideas(their_name: str) -> List[dict]:
    # TODO: Embed urls
    _header, *rows = get_rows(their_name)
    presents = [{**row, "index": index} for index, row in enumerate(rows)]
    return presents


def get_row_index(thing_index: int) -> int:
    # TODO: Why is it plus 3?!
    return thing_index + 3
