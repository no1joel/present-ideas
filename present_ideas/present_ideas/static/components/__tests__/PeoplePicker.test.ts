import { render } from "../../pages/__tests__/utils";
import PeoplePicker from "../PeoplePicker.vue";

describe("PeoplePicker", () => {
  it("matches snapshot", () => {
    expect(render(PeoplePicker).container).toMatchSnapshot();
  });
  it.todo("contains all the people's names");
});
