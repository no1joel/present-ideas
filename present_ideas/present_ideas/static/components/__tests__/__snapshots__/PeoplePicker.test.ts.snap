// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingIndicator matches snapshot 1`] = `
<div>
  <div
    class="d-flex justify-content-center align-items-center"
    style="background: rgba(255, 255, 255, 0.4); position: absolute; top: 0px; bottom: 0px; left: 0px; right: 0px;"
  >
    <div
      class="spinner-border"
      role="status"
    >
      <span
        class="sr-only"
      >
        Loading...
      </span>
    </div>
  </div>
</div>
`;

exports[`PeoplePicker matches snapshot 1`] = `
<div>
  <select
    class="custom-select"
  >
    <!---->
     
    <option
      disabled="disabled"
      value=""
    >
      
    Pick a name!
  
    </option>
     
  </select>
</div>
`;
